src/adapters/error-handler.js

```diff
@@ -0,0 +1,22 @@
+import { UnknownError } from "../lib/errors"
+
+/**
+ * Handles adapter induced errors.
+ * @param {import("types/adapters").AdapterInstance} adapter
+ * @return {import("types/adapters").AdapterInstance}
+ */
+export default function adapterErrorHandler(adapter) {
+  return Object.keys(adapter).reduce((acc, method) => {
+    const adapterMethod = adapter[method]
+    acc[method] = async (...args) => {
+      try {
+        return await adapterMethod(...args)
+      } catch (error) {
+        const e = new UnknownError(error)
+        e.name = `${method[0].toUpperCase()}${method.slice(1)}Error`
```

Comment:
turns out custom errors may not need to be defined as they only have  different name. this works great!